apps/ewallet_db/lib/ewallet_db/api_key.ex

```diff
@@ -94,84 +97,106 @@ defmodule EWalletDB.APIKey do
   Build the query for all APIKeys excluding the soft-deleted ones.
   """
   def query_all do
-    exclude_deleted(APIKey)
+    __MODULE__
+    |> exclude_deleted()
   end
 
   @doc """
-  Get API key by id, exclude soft-deleted.
+  Get an API key by id, exclude soft-deleted.
   """
-  @spec get(String.t()) :: %APIKey{} | nil
+  @spec get(String.t()) :: %__MODULE__{} | nil
   def get(id) when is_external_id(id) do
-    APIKey
+    __MODULE__
     |> exclude_deleted()
     |> Repo.get_by(id: id)
   end
 
   def get(_), do: nil
 
+  @doc """
+  Get an API key by a specific field, exclude soft-deleted.
+  """
+  @spec get_by(Keyword.t()) :: %__MODULE__{} | nil
+  def get_by(fields, opts \\ []) do
+    __MODULE__
+    |> exclude_deleted()
+    |> Repo.get_by(fields)
+    |> preload_option(opts)
+  end
+
   @doc """
   Creates a new API key with the passed attributes.
+
   The key is automatically generated if not specified.
   """
+  @spec insert(map()) :: {:ok, %__MODULE__{}} | {:error, Ecto.Changeset.t()}
   def insert(attrs) do
     attrs =
       attrs
-      |> Map.put_new_lazy(:account_uuid, fn -> get_master_account_uuid() end)
       |> Map.put_new_lazy(:key, fn -> Crypto.generate_base64_key(@key_bytes) end)
+      # Disallow overriding :creator_user_uuid and :creator_key_uuid from the request
+      |> Map.drop([:creator_user_uuid, :creator_key_uuid])
+      |> populate_creator()
 
-    %APIKey{}
+    %__MODULE__{}
     |> changeset(attrs)
     |> Repo.insert_record_with_activity_log()
   end
 
+  defp populate_creator(%{originator: %User{uuid: uuid}} = attrs) do
```

Comment:
Not sure if we should worry about this here. Usually, we don't handle this logic in the schemas (see `Key` where you can pass an access key / secret key if you want to), but at the controller/gate level.